File: app/src/main/java/de/xikolo/jobs/ListItemsWithContentForSectionJob.java
package de.xikolo.jobs;

import android.util.Log;

import com.birbit.android.jobqueue.Params;

import de.xikolo.config.Config;
import de.xikolo.jobs.base.BaseJob;
import de.xikolo.jobs.base.JobCallback;
import de.xikolo.models.base.Sync;
import de.xikolo.managers.UserManager;
import de.xikolo.models.Item;
import de.xikolo.network.ApiService;
import de.xikolo.utils.NetworkUtil;
import retrofit2.Response;

public class ListItemsWithContentForSectionJob extends BaseJob {

    public static final String TAG = ListItemsWithContentForSectionJob.class.getSimpleName();

    private String sectionId;

    public ListItemsWithContentForSectionJob(JobCallback callback, String sectionId) {
        super(new Params(PRIORITY_HIGH), callback);

        this.sectionId = sectionId;
    }

    @Override
    public void onAdded() {
        if (Config.DEBUG) {
            Log.i(TAG, TAG + " added | section.id " + sectionId);
        }
    }

    @Override
    public void onRun() throws Throwable {
        if (NetworkUtil.isOnline()) {
            if (UserManager.isAuthorized()) {

                Response<Item.JsonModel[]> response =
                        ApiService.getInstance().listItemsWithContentForSection(sectionId).execute();

                if (response.isSuccessful()) {
                    if (Config.DEBUG) Log.i(TAG, "Items received");

                    Sync.Data.with(Item.class, response.body())
                            .addFilter("sectionId", sectionId)
                            .run();
                    syncItemContent(response.body());

                    if (callback != null) callback.success();
                } else {
                    if (Config.DEBUG) Log.e(TAG, "Error while fetching section list");
                    if (callback != null) callback.error(JobCallback.ErrorCode.ERROR);
                }
            } else {
                if (callback != null) callback.error(JobCallback.ErrorCode.NO_AUTH);
            }
        } else {
            if (callback != null) callback.error(JobCallback.ErrorCode.NO_NETWORK);
        }
    }

}

File: app/src/main/java/de/xikolo/jobs/ListItemsWithContentForSectionJob.kt
package de.xikolo.jobs

import android.util.Log
import de.xikolo.config.Config
import de.xikolo.jobs.base.RequestJobCallback
import de.xikolo.jobs.base.RequestJob
import de.xikolo.models.Item
import de.xikolo.models.base.Sync
import de.xikolo.network.ApiService
import ru.gildor.coroutines.retrofit.awaitResponse

class ListItemsWithContentForSectionJob(callback: RequestJobCallback, private val sectionId: String) : RequestJob(callback, Precondition.AUTH) {

    companion object {
        val TAG: String = ListItemsWithContentForSectionJob::class.java.simpleName
    }

    override suspend fun onRun() {
        val response = ApiService.getInstance().listItemsWithContentForSection(sectionId).awaitResponse()

        if (response.isSuccessful) {
            if (Config.DEBUG) Log.i(TAG, "Items received")

            Sync.Data.with(Item::class.java, *response.body()!!)
                    .addFilter("sectionId", sectionId)
                    .run()
            syncItemContent(response.body()!!)

            callback?.success()
        } else {
            if (Config.DEBUG) Log.e(TAG, "Error while fetching section list")
            callback?.error(RequestJobCallback.ErrorCode.ERROR)
        }
    }

}
